src/jumps.jl

```diff
@@ -55,7 +55,7 @@ struct MassActionJump{T,S} <: AbstractJump
     end
     new(sr, rs, ns)
   end
-  function MassActionJump{T,S}(rate::T, rs_in::S, ns::S, scale_rates::Bool) where {T <: Number,S}    
+  function MassActionJump{T,S,U}(rate::T, rs_in::S, ns::U, scale_rates::Bool) where {T <: Number, S, U}    
```

Comment:
Get rid of the `{T,S,U}`.

Reply:
I think those are necessary; without them I get:

```
ERROR: LoadError: LoadError: syntax: too few type parameters specified in "new{...}"
Stacktrace:
 [1] include_from_node1(::String) at ./loading.jl:576
 [2] include(::String) at /Applications/Julia-0.6.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 [3] include_from_node1(::String) at /Applications/Julia-0.6.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 [4] include(::String) at /Applications/Julia-0.6.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 [5] anonymous at ./<missing>:2
while loading /Users/isaacsas/.julia/v0.6/DiffEqJump/src/jumps.jl, in expression starting on line 39
```
...

The problem was that I wanted to have an optional named parameter, `scale_rates`, for which I added the outer constructor. We want the outer constructor to always be called, even if the user passes only the three required parameters (which is what the default inner constructor would take). As far as I can tell, the explicit typing in the inner function definitions, and the call to the explicitly typed form from the outer constructor, are the only way to avoid the outer constructor being shadowed. 

If you know an alternative way to set this up I'm happy to change it.

Reply:
Oh that's my bad. From the diff I didn't see it was an inner constructor.